File: lib/models/Card.dart
import 'package:flutter/foundation.dart';

class Card {
  late String cardNumber;
  late String expiryDate ;
  late String cardHolderName;
  late int cvvCode;
  late int pinNo;
  late String cardType;

  Card({
    required this.cardNumber,
    required this.expiryDate,
    required this.cardHolderName,
    required this.cvvCode,
    required this.pinNo,
    required this.cardType,
  });

  factory Card.fromJson(Map<String, dynamic> json) {
    return Card(
        cardNumber: json['cardNumber'],
        expiryDate: json['expiryDate'],
        cardHolderName: json['cardHolderName'],
        cvvCode: json['cvvCode'],
        cardType: json['cardType'],
        pinNo: json['pinNo']);
  }

  Map<String, dynamic> toJson() => {
    'cardNumber': cardNumber,
    'expiryDate': expiryDate,
    'cardHolderName': cardHolderName,
    'cvvCode': cvvCode,
    'cardType': cardType,
    'pinNo': pinNo
  };
}

File: save_my_pin_be/models/Card.js
const mongoose = require('mongoose');
const  Schema = mongoose.Schema;

const cardSchema = new Schema({

    cardNumber: {
        type: String,
        required: true
    },
    expiryDate: {
        type:String,
        required: true
    },
    cardHolderName: {
        type: String, 
        required: true
    },
    cvvCode: {
        type: Number, 
        required: true, 
    },
    cardType:{
        type: String, 
        required: true, 
    },
    pinNo: {
        type: Number,
        required: true
    },
    owner: {
        type: Schema.Types.ObjectId, ref: 'User',
        required: true,
    }

})

const Card = mongoose.model("Card", cardSchema);
module.exports = Card;